### DISCUSSION TITLE
Screenshot of model viewer div... Web site (Capture Canvas)

### DISCUSSION THREAD

**Discussion by canberka** [google/model-viewer#3229] [Q&A]
### Description

Hello, we are trying to screenshot of model viever at website.

But when we are trying to screenshot scripts for model viever. it can takes  screenshot of everything except model viever.

Is there a way to handle with it?

[[Link](https://karaca.basebros.com/customize-test2.html)]

if you try to link, you will see. if you press the button it is not capturing the model viewer div

**Comment by samaneh-kazemi** [comment]
Hi @canberka, can you please share more details? A link to your work, in what mode you're using model-viewer, is this happening in iPhone, android etc?

**Comment by canberka** [comment]
> Hi @canberka, can you please share more details? A link to your work, in what mode you're using model-viewer, is this happening in iPhone, android etc?

I am talking about capturing model viewer div. I updated the description. Can you please check

**Comment by sercandemirtas** [comment]
Hey @canberka Im having same problem too. I cant get capture of model-viewer div or canvas. I tried 4 different capture script. All same like your link. On capture of div or canvas, model-viewer part is blank.

**Comment by elalish** [comment]
That looks like a bug in `html2canvas`? I'm not familiar. However, if what you want is a screenshot of MV, we have a built-in function for that: https://modelviewer.dev/docs/index.html#entrydocs-loading-methods-toBlob

**Reply by yi-ming-qian** [reply]
How to use this toBlob function? Any example codes will be much appreciated.

**Reply by base-main-online** [reply]
`document.querySelector("model-viewer").toDataURL("image/png")=img.src`

**Reply by elalish** [reply]
Use it just like the canvas one: https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob

**Comment by arpu** [comment]
the toDataUrl works great! but would it be possible to generate a gif like webp or video?

**Reply by elalish** [reply]
You'll just need to take a bunch of screenshots and then pack them into a movie with ffmpeg or similar. Or just do a screen recording.

**Reply by arpu** [reply]
yes this was the idea, maybe like threejs will do it https://github.com/mrdoob/three.js/pull/25296
